validator/likecoin-chain-node/setup-a-node.md

```diff
@@ -1,108 +1,440 @@
+
+- [UFW usage](https://www.digitalocean.com/community/tutorials/how-to-set-up-a-firewall-with-ufw-on-ubuntu-20-04)
+- [Generate ed25519 SSH Key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
+- [Deploy SSH key to server](https://www.digitalocean.com/community/tutorials/how-to-configure-ssh-key-based-authentication-on-a-linux-server#step-2-copying-an-ssh-public-key-to-your-server)
+- [SSH Hardening](https://www.digitalocean.com/community/tutorials/how-to-harden-openssh-client-on-ubuntu-20-04)
+
+The physical security of the host should be reviewed as well.
+
+## Systemd Route (Recommended)
+
+We will run a setup script to download the latest binary release, write config files and start up the service.
+
+Again, this method is supported on Linux only. We prepared this guide with Ubuntu 20.04 LTS. YMMV on other distros.
+
+> When you see `<Placeholder>` in code blocks below, please replace them with correct values.
+
+### Clone likecoin-chain repo
+
+We will first clone the likecoin-chain repo to obtain required scripts
+
+> For validator migration, we recommend cloning a new copy to avoid confusion, since the previous guide stored data within the repo folder.
+>
+> If your existing repo folder is already at `~`, appending the clone command with a new folder name, and adopt this guide by changing the `cd ~/likecoin-chain` commands.
+
+```shell
+cd ~
+git clone https://github.com/likecoin/likecoin-chain.git --branch master --single-branch
+```
+
+### Run the setup-node script
+
+This script will download cosmovisor, chain binary, and the genesis file. Then, it will run `liked init` and prepare a systemd service file.
+
+By default, binaries will be downloaded to the user's home directory `~/`, and the liked data directory will be `~/.liked`
+
+For advanced configuration, such as changing folder locations or specifying binary versions, refer to [the developer facing guide](https://github.com/likecoin/likecoin-chain/blob/master/SETUP.md)
+
+> If you are migrating an existing node on the same host, please ensure that your existing `.liked` folder is not at your home directory, otherwise this might overwrite your files!
+
+For latest genesis url and seed node, refer to:
+
+- [Mainnet](https://github.com/likecoin/mainnet/)
+- [Public Testnet](https://github.com/likecoin/testnets)
+
+```shell
+cd ~/likecoin-chain
+export MONIKER='My Validator'
+export GENESIS_URL='https://raw.githubusercontent.com/likecoin/mainnet/master/genesis.json'
+export LIKED_SEED_NODES='913bd0f4bea4ef512ffba39ab90eae84c1420862@34.82.131.35:26656,e44a2165ac573f84151671b092aa4936ac305e2a@nnkken.dev:26656'
+make -C deploy/cmd/deploy/assets setup-node
+```
+
+### Import existing validator files
+
+> For migrating existing validators only. New operators should skip this section.
+
+First, rename the `.liked` folder generated by the script as `.liked.template`:
```

Comment:
For existing validator using docker-compose setting following previous suggested docker-compose setup.

Reply:
updated